src/sentry/sentry_apps/external_requests/issue_link_requester.py

```diff
@@ -0,0 +1,129 @@
+import logging
+from dataclasses import dataclass
+from typing import Any
+from urllib.parse import urlparse
+from uuid import uuid4
+
+from django.db import router, transaction
+from django.utils.functional import cached_property
+
+from sentry.coreapi import APIError
+from sentry.http import safe_urlread
+from sentry.models.group import Group
+from sentry.sentry_apps.external_requests.utils import send_and_save_sentry_app_request, validate
+from sentry.sentry_apps.services.app import RpcSentryAppInstallation
+from sentry.users.services.user import RpcUser
+from sentry.utils import json
+
+logger = logging.getLogger("sentry.sentry_apps.external_requests")
+ACTION_TO_PAST_TENSE = {"create": "created", "link": "linked"}
+
+
+@dataclass
+class IssueLinkRequester:
+    """
+    1. Makes a POST request to another service with data used for creating or
+    linking a Sentry issue to an issue in the other service.
+
+    The data sent to the other service is always in the following format:
+        {
+            'installationId': <install_uuid>,
+            'issueId': <sentry_group_id>,
+            'webUrl': <sentry_group_web_url>,
+            <fields>,
+        }
+
+    <fields> are any of the 'create' or 'link' form fields (determined by
+    the schema for that particular service)
+
+    2. Validates the response format from the other service and returns the
+    payload.
+
+    The data sent to the other service is always in the following format:
+        {
+            'identifier': <some_identifier>,
+            'webUrl': <external_issue_web_url>,
+            'project': <top_level_identifier>,
+        }
+
+    The project and identifier are use to generate the display text for the linked
+    issue in the UI (i.e. <project>#<identifier>)
+    """
+
+    install: RpcSentryAppInstallation
+    uri: str
+    group: Group
+    fields: dict[str, Any]
+    user: RpcUser
+    action: str
+
+    def run(self) -> dict[str, Any]:
+        with transaction.atomic(router.db_for_write(Group)):
```

Comment:
I dont think we need this transaction ? It doesn't look like we're making an db requests here or downstream unless the callers of this are ? but in that case wouldn't that be the caller's responsibility to wrap their call in a transaction?

Reply:
Yeah, I'm not seeing and DB operations in this or the called methods. We could skip the transaction here.